sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastWithAnsiOnSuite.scala

```diff
@@ -797,4 +797,25 @@ class CastWithAnsiOnSuite extends CastSuiteBase with QueryErrorsBase {
         castErrMsg(invalidInput, TimeType()))
     }
   }
+
+  test("cast TimeType to integral types with overflow with ansi on") {
+    // Test overflow cases: 23:59:59 = 86399 seconds
+    val largeTime6 = Literal.create(LocalTime.of(23, 59, 59), TimeType(6))
+    val largeTime4 = Literal.create(LocalTime.of(23, 59, 59), TimeType(4))
+
+    // Short and Byte should overflow and throw ArithmeticException (ANSI mode)
+    // 86399 > Short.MaxValue (32767) and > Byte.MaxValue (127)
+    checkExceptionInExpression[SparkArithmeticException](
```

Comment:
Could you use `checkErrorInExpression` instead of the legacy `checkExceptionInExpression`, please. Error messages are not stable, and tech writers can change them. The tests shouldn't depend on error message formats. Ideally only from error message parameters.

Reply:
I actually wondered if it seemed brittle too but I wasn't sure. Thanks for telling me about the `checkErrorInExpression`